crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -377,7 +377,7 @@ impl Format<PyFormatContext<'_>> for FormatTrailingEndOfLineComment<'_> {
         let normalized_comment = normalize_comment(self.comment, source)?;
 
         // Don't reserve width for excluded pragma comments.
-        let reserved_width = if is_pragma_comment(&normalized_comment) {
+        let reserved_width = if is_pragma_comment(&normalized_comment).0 {
             0
         } else {
```

Comment:
I'd expect it to account for the width up to but not including the start of the pragma comment. E.g. it should include the width of the first part in `# this counts to the width # noqa: RUF` 

It's important the logic here is consistent with the logic in the linter. We should add tests to ensure this is the case.